alchemy/src/context.ts

```diff
@@ -53,7 +53,7 @@ export interface BaseContext<Out extends Resource> {
    * Indicate that this resource is being replaced.
    * This will cause the resource to be deleted at the end of the stack's CREATE phase.
    */
-  replace(): void;
+  replace(): Promise<void>;
```

Comment:
`Promise<never>`?

Reply:
Or just `never` - the async write to state can be done outside of the replace handler but still durably before calling `create` for the new resource

Reply:
done